### PR TITLE
Deprecate the tap_common directory and makefile references to tap-v1.

### PR DESCRIPTION
**Pull request by mcsauder**
Hello @potaito and @simonegu ,

This PR deprecates the `ROMFS/tap_common/` directory per the discussion in PR #10532 and also deprecates all makefile references to `tap-v1*`.

Please let me know if there are other deletions that should be accomplished within this PR, or if I've deleted anything that should not have been eliminated given the objectives of this PR.

Thanks!

-Mark

### CODE REVIEW

**Comment by potaito** [comment]
You probably need to remove `tap-v1` from jenkins :)

**Comment by dagar** [comment]
https://github.com/PX4/Firmware/pull/10187

**Comment by mcsauder** [comment]
Thanks @dagar! I have removed the additional files indicated in your PR.

File: .ci/Jenkinsfile-compile [hunk 1]
```diff
@@ -89,7 +89,7 @@ pipeline {
           def nuttx_builds_archive = [
             target: ["px4fmu-v2_default", "px4fmu-v3_default", "px4fmu-v4_default", "px4fmu-v4pro_default", "px4fmu-v5_default", "px4fmu-v5_rtps", "px4fmu-v5_stackcheck",
                      "aerofc-v1_default", "aerocore2_default", "auav-x21_default", "av-x-v1_default", "crazyflie_default", "mindpx-v2_default",
-                     "nxphlite-v3_default", "tap-v1_default", "omnibus-f4sd_default"],
+                     "nxphlite-v3_default", "omnibus-f4sd_default"],
```
Comment (mcsauder) [inline .ci/Jenkinsfile-compile]: I had thought that this line would pass the Jenkins CI pipeline with the deprecated files, although perhaps it does not reference the `.ci/` directory changes for running the tests.  Please let me know if I have overlooked anything in this regard!

**Comment by mcsauder** [comment]
@dagar, would you be willing to help me with a hint on this CI error?

image

**Comment by potaito** [comment]
Hm it's still trying to build `tap-v1_default`
image
even though @mcsauder removed it from the jenkins file :confused:

**Comment by lamping7** [comment]
@mcsauder @potaito Jenkins only trusts executing the jenkinsfile from the origin repo. Because your PR is not a branch in the PX4 repo, rather it is from your fork, it will ignore the changes to the Jenkinsfile until merged over.

**Comment by mcsauder** [comment]
Thanks @lamping7 , that makes sense.

@potaito or @simonegu , I don't have privileges to push branches up to the repo, would one of you be willing to check out my branch and push up a new branch branch/PR to PX4/Firmware from it?

Let me know what you need from me!

**Comment by potaito** [comment]
@dagar Ah I did not realize that it's not clear. It's fine to remove everything.

**Comment by mcsauder** [comment]
Thanks everyone!

**Pull request merged**